cpp/models/lct_secir/parameters_io.cpp

```diff
@@ -0,0 +1,275 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "lct_secir/parameters_io.h"
+#include "memilio/config.h"
+
+#ifdef MEMILIO_HAS_JSONCPP
+
+#include "lct_secir/infection_state.h"
+#include "lct_secir/parameters.h"
+#include "memilio/io/epi_data.h"
+#include "memilio/io/io.h"
+#include "memilio/utils/date.h"
+#include "memilio/utils/logging.h"
+#include "memilio/math/eigen.h"
+
+#include <string>
+
+namespace mio
+{
+namespace lsecir
+{
+
+IOResult<Eigen::VectorXd> get_initial_data_from_file(std::string const& path, Date date, InfectionState infectionState,
```

Comment:
I find get_initial_data_from_file not a fitting name for the function as you are not only getting the data but you are computing the initial data from the case data. Maybe compute_initial_data_from_confirmed_cases instead?

Reply:
I had to change the whole structure so that the model is now an input parameter and the initial value vector is set directly in the model. Accordingly, I have named the function set_initial_data_from_confirmed_cases.